File: turing/test2.py
# Given a binary tree [3,9,20,null,null,15,7],
#
#     3
#    / \
#   9  20
#     /  \
#    15   7
# return its depth = 3.
#
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
#
# class Solution:
#     def maxDepth(self, root: TreeNode) -> int:
#         if root is None:
#             return 0
#         return max(self.maxDepth(root.left), self.maxDepth(root.right)) + 1
#
# if __name__ == "__main__":
#     line = input()
#     root = TreeNode(int(line))
#     line = input()
#     root.left = TreeNode(int(line))
#     line = input()
#     root.right = TreeNode(int(line))
#     line = input()
#     root.left.left = TreeNode(int(line))
#     line = input()
#     root.left.right = TreeNode(int(line))
#     line = input()
#     root.right.left = TreeNode(int(line))
#     line = input()
#     root.right.right = TreeNode(int(line))
#     print(Solution().maxDepth(root))
#
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root is None:
            return 0
        return max(self.maxDepth(root.left), self.maxDepth(root.right)) + 1

if __name__ == "__main__":
    line = input()
    root = TreeNode(int(line))
    line = input()
    root.left = TreeNode(int(line))
    line = input()
    root.right = TreeNode(int(line))
    line = input()
    root.left.left = TreeNode(int(line))
    line = input()
    root.left.right = TreeNode(int(line))
    line = input()
    root.right.left = TreeNode(int(line))
    line = input()
    root.right.right = TreeNode(int(line))
    print(Solution().maxDepth(root))

result = Solution().maxDepth('[3,9,20,null,null,15,7]')
print(result)

File: turing/test2.js
/**
 * @param TreeNode root
 * @return int
 */
// Given a binary tree, find its maximum depth.
// The maximum depth is the number of nodes along the longest path from the root node down to the farthest leaf node.
// Note: A leaf is a node with no children.
// Example:
// Given binary tree [3,9,20,null,null,15,7],
//     3
//    / \
//   9  20
//     /  \
//    15   7
// return its depth = 3.

function maxDepth(root) { 
    if (root === null) {
        return 0;
    }
    var maxDepth = 0;
    var stack = [];
    stack.push(root);
    while (stack.length !== 0) {
        var node = stack.pop();
        if (node.left !== null) {
            stack.push(node.left);
        }
        if (node.right !== null) {
            stack.push(node.right);
        }
        if (node.left === null && node.right === null) {
            maxDepth++;
        }
    }
    return maxDepth;
}

// x = maxDepth('3,9,20,null,null,15,7');
// console.log(x);
